### PR TITLE
port number from command line

### PR DESCRIPTION
**Pull request by apoorvemohan**
changes for a port being passed as a command line argument. please review.

### CODE REVIEW

**Comment by zenhack** [comment]
Looks good to me, +1.

Ideally we'd have it choose a default, and be overridable, but that's going to require some refactoring of the cli tool.

@gsilvis, @lokI8, @henn, can one of you do a second review?

**Comment by apoorvemohan** [comment]
exception handling for invalid port number committed

File: haas/cli.py [hunk 1]
```diff
@@ -27,6 +27,8 @@
 
 command_dict = {}
 usage_dict = {}
+MIN_PORT_NUMBER = 0
```
Comment (zenhack) [inline haas/cli.py]: Port 0 has special meaning:

http://lxr.free-electrons.com/source/net/ipv4/inet_connection_sock.c?v=3.18#L89

Start with port 1.

File: haas/cli.py [hunk 2]
```diff
@@ -27,6 +27,8 @@
...
+MAX_PORT_NUMBER = 65525
```
Comment (zenhack) [inline haas/cli.py]: How about `2**16 - 1`? makes it a bit more clear where the number is coming from.

File: haas/cli.py [hunk 3]
```diff
@@ -89,7 +91,18 @@ def do_delete(url):
...
 @cmd
-def serve():
+def serve(port):
+    isNotValidPort = False
+    if port.isdigit():
+        port = int(port)
+        if (port < MIN_PORT_NUMBER) or (port > MAX_PORT_NUMBER):
+            isNotValidPort = True
+    else:
+	isNotValidPort = True
```
Comment (zenhack) [inline haas/cli.py]: Check out the schema library (we're using it elsehwere in the code) --- I bet you can get this down to a (readable) one-liner.

File: haas/cli.py (hunk 1 above)
Comment (zenhack) [inline haas/cli.py]: Ah, had another look. You're using strictly less than in the check so this works actually, but in my mind "MIN" implies an inclusive range, so the name is a bit confusing.

**Comment by apoorvemohan** [comment]
changes committed

File: haas/cli.py [hunk 4]
```diff
@@ -89,7 +92,12 @@ def do_delete(url):
     return check_status_code(requests.delete(url))
 
 @cmd
-def serve():
+def serve(port):
+    try:
+        port = schema.And(schema.Use(int), lambda n: MIN_PORT_NUMBER <= n <= MAX_PORT_NUMBER).validate(port)
+    except:
```
Comment (zenhack) [inline haas/cli.py]: Always specify the actual error you're expecting --- in this case `schema.SchemaError`. If some error you're not expecting occurs, you want to hear about it (and not necessarily handle it the same way).

File: haas/cli.py [hunk 5]
```diff
@@ -89,7 +92,14 @@ def do_delete(url):
     return check_status_code(requests.delete(url))
 
 @cmd
-def serve():
+def serve(port):
+    try:
+        port = schema.And(schema.Use(int), lambda n: MIN_PORT_NUMBER <= n <= MAX_PORT_NUMBER).validate(port)
+    except schema.SchemaError:
+	sys.exit('Error: Invaid port. Must be in the range 1-65535.')
+    except as e:
```
Comment (zenhack) [inline haas/cli.py]: You need a type here; `except Exception as e` is probably what you right. As it stands, this is a syntax error.

**Comment by zenhack** [comment]
Okay, this seems to be working. @gsilvis, this *does* break backwards compatibility, but I'd like to just not worry about it --- it's in a dev tool and I don't want to imply any promises unless we need to. Thoughts? I'm otherwise happy with this, +1

**Comment by lokI8** [comment]
+1, merging. Ian's right that we need optional arguments in the CLI

**Pull request merged**